frontend/src/views/sql-editor/EditorCommon/ResultView/DataTable/binary-format-store.ts

```diff
@@ -0,0 +1,126 @@
+import { ref } from 'vue';
+
+// Store for binary data formatting
+// This store maps row/column coordinates to their selected format
+type FormatMap = Map<string, string>;
+type ColumnTypeMap = Map<string, string>;
+type ColumnFormatMap = Map<string, string>;
+
+// Create a global format map using Vue's ref for reactivity
+const formattedBinaryValues = ref<FormatMap>(new Map());
+
+// Store column types to help with auto-detection
+const columnTypes = ref<ColumnTypeMap>(new Map());
+
+// Store column format overrides
+const columnFormatOverrides = ref<ColumnFormatMap>(new Map());
+
+// Get a unique key for a cell's row/column position
+const getCellKey = (rowIndex: number, colIndex: number, tableIndex = 0): string => {
+  return `${tableIndex}:${rowIndex}:${colIndex}`;
+};
+
+// Get a key for a column
+const getColumnKey = (colIndex: number, tableIndex = 0): string => {
+  return `${tableIndex}:col:${colIndex}`;
+};
+
+// Store a format for a specific cell
+export const setBinaryFormat = (
```

Comment:
I think it's better to use `setIndex`, `rowIndex` and `colIndex` which consistent with the names in components. And as there're too many number inputs, in order to avoid mistake in the future, we'd better use object params like

```
const setBinaryFormat = (params: {
setIndex: number;
rowIndex: number;
colIndex: number;
format: string;
}) => {}
```

Reply:
Same for other functions